src/rules/padding.ts

```diff
@@ -0,0 +1,384 @@
+/**
```

Comment:
Can you highlight the things in this file that differ from the original implementation in `eslint/padding-line-between-statements`

Reply:
Yep. (I'm not sure how good/useful my description will be, so you may also want to do a `diff -by --color=always node_modules/eslint/lib/rules/padding-line-between-statements.js src/rules/padding.ts` in this branch to see the differences for yourself.)

Here are the main changes, though.
- Added a `newJestTokenTester` function
- Added an `isTestFile` function
- Modified `StatementTypes` to look for Jest tokens. 
  This allowed removal of several now unused functions.
- Modified `PaddingTypes` to remove `never`
  This also allowed removal of several now unused functions.
- Removed `meta.docs` from the rule per the ESLint docs on working with rules
  > In a custom rule or plugin, you can omit docs or include any properties that you need in it.

Other than those, the differences are a bunch of formatting changes from prettier.

I'll make this file a little more TypeScripty with one of my next few commits.

Reply:
TypeScripting it up. Almost done.